pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -249,7 +284,27 @@ func (p *DNSProxy) RestoreRules(ep *endpoint.Endpoint) {
 	if ep.IPv6.IsSet() {
 		p.restoredEPs[ep.IPv6.String()] = ep
 	}
-	p.restored[uint64(ep.ID)] = ep.DNSRules
+	restoredRules := make(map[uint16][]restoredIPRule, len(ep.DNSRules))
+	for port, dnsRule := range ep.DNSRules {
+		ipRules := make([]restoredIPRule, 0, len(dnsRule))
+		for _, ipRule := range dnsRule {
+			var regex *regexp.Regexp
+			if ipRule.Re.Pattern != nil {
+				var err error
+				regex, err = p.cache.lookupOrCompileRegex(*ipRule.Re.Pattern)
+				if err != nil {
+					log.Warnf("Error compiling regex for endpointID %d, gracefully ignoring", ep.ID)
```

Comment:
I think `INFO` level is more appropriate for this log msg. I think we'll need a more user-friendly msg as well.

```suggestion
					log.WithFields(logrus.Fields{
					  logfields.EndpointID: ep.ID,
					  logfields.Rule: *ipRule.Re.Pattern,
					}).Info("Disregarding restored DNS rule due to failure in compiling regex. Traffic to the FQDN may be disrupted.")
```